lte/gateway/python/magma/health/state_recovery.py

```diff
@@ -0,0 +1,118 @@
+    sctpd as recovery method for services crashing
+    """
+
+    def __init__(self, service_state: ServiceStateWrapper,
+                 polling_interval: int, services_check: List[str],
+                 restart_threshold: int, redis_dump_src: str,
+                 snapshots_dir: str, service_loop):
+        super().__init__(interval=polling_interval, loop=service_loop)
+        self._state_wrapper = service_state
+        self._services_check = services_check
+        self._threshold = restart_threshold
+        self._snapshots_dir = snapshots_dir
+        self._redis_dump_src = redis_dump_src
+        self._loop = service_loop
+        self._polling_interval = polling_interval
+        self._services_restarts_map = defaultdict(int)
+
+    def _get_service_status(self, service_name: str) \
+            -> Optional[ServiceExitStatus]:
+        """
+        Args:
+            service_name: name of magma service to check status
+
+        Returns: ServiceExitStatus obj for given service
+
+        """
+        try:
+            service_status = self._state_wrapper.get_service_status(
+                service_name)
+            return service_status
+        except (KeyError, ConnectionError) as err:
+            logging.debug('Could not obtain service status: [%s]' % err)
+            return None
+
+    async def restart_service_async(self, service: str):
+        """
+        Execute service restart commands asynchronously.
+        """
+        await subprocess_workflow.exec_and_parse_subprocesses_async(
+            [SystemdServiceParams(service)],
+            _get_service_restart_args,
+            None,
+            self._loop,
+        )
+
+    async def _run(self):
+        for service in self._services_check:
+            result = self._get_service_status(service)
+            if result:
+                last_num_restarts = self._services_restarts_map[service]
+                current_restarts = result.num_fail_exits - last_num_restarts
+                if current_restarts > self._threshold:
+                    logging.info(
+                        'Service %s has failed %s times over last %s seconds, '
+                        'restarting sctpd to clean state...', service,
+                        current_restarts,
+                        self._polling_interval)
+
+                    # Save RDB snapshot
+                    os.makedirs(os.path.dirname(self._snapshots_dir),
+                                exist_ok=True)
+                    shutil.copy("%s/redis_dump.rdb" % self._redis_dump_src,
+                                "%s/redis_dump_%s.rdb" % (
+                                    self._snapshots_dir, time()))
+
+                    # Restart sctpd
+                    await self.restart_service_async('sctpd')
+
+                # Update latest number of restarts for service
+                self._services_restarts_map[service] = current_restarts
```

Comment:
The _service_restarts_map should be updated to the latest value read from Redis.
```suggestion
                self._services_restarts_map[service] = result.num_fail_exits
```

Reply:
> @ardzoht can you also include the "corrupt" functionality you added in state_cli.py to test these changes.
> 
> We can use it to add an S1ap integration test for this recovery service.

@ssanadhya Included it on the PR